templates/power-off-vm.json

```diff
@@ -7,46 +7,79 @@
+                                        "host": {
+                                            "connection": {
+                                                "name": "@parameters('$connections')['office365']['connectionId']"
+                                            }
+                                        },
+                                        "method": "post",
+                                        "path": "/v2/Mail"
                                     },
-                                    "host": {
-                                        "connection": {
-                                            "name": "@parameters('$connections')['office365']['connectionId']"
-                                        }
-                                    },
-                                    "method": "post",
-                                    "path": "/v2/Mail"
-                                },
-                                "runAfter": {},
-                                "type": "ApiConnection"
-                            }
+                                    "runAfter": {},
+                                    "type": "ApiConnection"
+                                }
+                            },
+                            "else": {
+                                "actions": {
+                                    "Send_an_email_(V2)": {
+                                        "inputs": {
+                                            "body": {
+                                                "Body": "<p><strong>FAILED <br>\n<br>\n</strong>Status: @{body('Get_virtual_machine')?['properties']?['provisioningState']}</p>",
+                                                "Subject": "Your automation task failed to stop virtual machine @{body('Get_virtual_machine_state')?['name']}",
+                                                "To": "@parameters('EmailID')"
+                                            },
+                                            "host": {
+                                                "connection": {
+                                                    "name": "@parameters('$connections')['office365']['connectionId']"
+                                                }
+                                            },
+                                            "method": "post",
+                                            "path": "/v2/Mail"
+                                        },
+                                        "runAfter": {},
+                                        "type": "ApiConnection"
+                                    }
+                                }
+                            },
+                            "expression": {
+                                "and": [
+                                    {
+                                        "equals": [
+                                            "@body('Get_virtual_machine')?['properties']?['provisioningState']",
```

Comment:
Is this a valid action "Get_virtual_machine" I dont see this action in template. I only see ""Get_virtual_machine_state"" as action

Reply:
it's there actually

Reply:
fixed the statement error